tests/transaction_gas_price_test.go

```diff
@@ -0,0 +1,352 @@
+package tests
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/require"
+)
+
+// constant large enough to satisfied tx validation checks
+// deduced from net rules minimum gas price
+const enoughGasPrice = 150_000_000_000
+
+func TestTransactionGasPrice(t *testing.T) {
+
+	net, err := StartIntegrationTestNet(t.TempDir())
+	require.NoError(t, err)
+	defer net.Stop()
+
+	client, err := net.GetClient()
+	require.NoError(t, err)
+	defer client.Close()
+
+	// use a fresh account to send transactions from
+	account := makeAccountWithBalance(t, net, int64(1e18))
+
+	t.Run("Legacy transaction, effectivePrice is equal to requested price", func(t *testing.T) {
+
+		// This test:
+		// 1. Compute a valid gas price
+		// 2. Sends a Legacy transaction with specified gas price
+		// 3. Checks
+		//    - effective gas price is greater or equal than basefee
+		//    - subtracted balance equals gas price * gas used + value transferred
+		//    - effective gas price equals the specified price
+
+		balanceBefore := getBalance(t, client, account.Address())
+
+		// 1: gas price
+		var specifiedPrice int64 = enoughGasPrice
+
+		// 2: make & execute transaction
+		tx := makeLegacyTx(t, client, specifiedPrice, account)
+
+		receipt, err := net.Run(tx)
+		require.NoError(t, err)
+		require.Equal(t,
+			receipt.Status,
+			types.ReceiptStatusSuccessful,
+			"transaction execution failed",
+		)
+
+		// 3: checks
+		t.Run("Transaction gas price >= BaseFee  ", func(t *testing.T) {
+			basefeeAfter := getBaseFee(t, client)
```

Comment:
This is correct, the test does not submit any transaction that could yield a block. But given the right timing, a new block could be produced in between. 
(this may be the issue I am experiencing with the parallel execution, time shifts and the last block is no longer the wanted one)
I will add this change.